spark-cassandra-connector/src/main/scala/org/apache/spark/metrics/CassandraConnectorSource.scala

```diff
@@ -4,24 +4,68 @@ import com.codahale.metrics
 import org.apache.spark.SparkEnv
 import org.apache.spark.metrics.source.Source
 
-object CassandraConnectorSource extends Source {
+/** This class is a Source implementation for Cassandra Connector related Codahale metrics. It is
+  * automatically instantiated and registered by Spark metrics system if this source is specified in
+  * metrics configuration file.
+  *
+  * Spark instantiates this class when `SparkEnv` is started. There can be only a single instance of
+  * `SparkEnv` so there can be at most a single active instance of `CassandraConnectorSource`.
+  * The active instance is assigned to [[CassandraConnectorSource._instance]] so that it can be retrieved
+  * from anywhere by [[CassandraConnectorSource.instance]] method. We need this because we have to access
+  * the meters from the task execution.
+  */
+class CassandraConnectorSource extends Source {
   override val sourceName = "cassandra-connector"
 
   override val metricRegistry = new metrics.MetricRegistry
 
-  val writeByteMeter = metricRegistry.meter("write-byte-meter")
-  val writeRowMeter = metricRegistry.meter("write-row-meter")
-  val writeBatchTimer = metricRegistry.timer("write-batch-timer")
+  val writeByteMeter      = metricRegistry.meter("write-byte-meter")
```

Comment:
Don't align like this. I know it looks more readable *now*, but on the other hand whenever you change a name of a field, the alignment gets broken. And you might not notice it when using automatic refactoring. And if do notice, and you fix alignment, you're only making probability of conflicts higher and adding a diff noise.